provider/metaschema/schema.go

```diff
@@ -96,6 +99,72 @@ func (s Schema) TypeAtTerraformPath(ctx context.Context, p *tftypes.AttributePat
 	return fwschema.SchemaTypeAtTerraformPath(ctx, s, p)
 }
 
+// Validate verifies that the schema is not using a reserved field name for a top-level attribute.
+func (s Schema) Validate() diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	// Raise error diagnostics when data source configuration uses reserved
+	// field names for root-level attributes.
+	reservedFieldNames := map[string]struct{}{
+		"alias":   {},
+		"version": {},
+	}
```

Comment:
Provider meta schemas relate to the [`terraform` configuration block `provider_meta`](https://developer.hashicorp.com/terraform/internals/provider-meta) so we can skip these `provider` configuration block specific name checks 👍 

```suggestion
```